e3sm_diags/parser/core_parser.py

```diff
@@ -815,21 +815,21 @@ def _get_cfg_parameters(
         parameters = []
 
         cfg_file_obj = self._create_cfg_hash_titles(cfg_file)
-        kwargs = (
-            {"strict": False} if sys.version_info[0] >= 3 else {}
-        )  # 'strict' keyword doesn't work in Python 2.
-        config = configparser.ConfigParser(  # type: ignore
-            **kwargs
-        )  # Allow for two lines to be the same.
-        config.readfp(cfg_file_obj)
-
-        for section in config.sections():
```

Comment:
`kwargs` is no longer necessary because e3sm_diags runs on Python >=3.9 right now.

Also updated comment to make it clearer why `strict=False`.